pkg/bundle/source.go

```diff
@@ -81,6 +81,12 @@ func (b *bundle) buildSourceBundle(ctx context.Context, sources []trustapi.Bundl
 			return bundleData{}, fmt.Errorf("failed to retrieve bundle from source: %w", err)
 		}
 
+		// A source selector may select no configmaps/secrets, and this is not an error.
+		// Must continue early since CertPool does not currently support adding no certificates.
+		if len(sourceData) == 0 {
+			continue
+		}
```

Comment:
suggestion: Would it be worth logging a warning in this situation? As a platform admin, I'd want to either remove a selector which selects nothing, or fix it.

Reply:
I agree that a platform admin needs to understand what the operator is doing (and not doing). It seems like events are used for similar scenarios in this project. Should we use an event for this too, or just log it? With contextual logging, I am always in doubt what you exactly mean with a "logging a warning". 😉

Reply:
I'd personally have just logged it, and wouldn't have thought about events. I don't think an event would be bad, but I think the lightweight option of logging is totally fine!

Reply:
I'm now logging if a label selector matches no resources. Not having log levels feels a bit strange to me, but supported by Dave Chaney, I am proposing to log this on a normal info level.

> Eliminate the warning level, it’s either an informational message, or an error condition.

Ref. https://dave.cheney.net/tag/logging

I also introduced a custom error to eliminate the risk of other errors slipping through the empty string conditional. PTAL!